cmd/shield/main.go

```diff
@@ -2635,25 +2633,55 @@ func main() {
 			opts.Tasks.Target = t.UUID
 		}
 
+		var timeBefore int64
+		if opts.Tasks.BeforeCreatedDate != "" {
+			timeBefore = strptime(opts.Tasks.BeforeCreatedDate)
+		} else {
+			timeBefore = time.Now().Unix()
+		}
+
 		filter := &shield.TaskFilter{
-			Status: opts.Tasks.Status,
-			Limit:  &opts.Tasks.Limit,
-			Target: opts.Tasks.Target,
+			Status:            opts.Tasks.Status,
+			Limit:             &opts.Tasks.Limit,
+			Target:            opts.Tasks.Target,
+			BeforeCreatedDate: timeBefore,
 		}
+
 		if opts.Tasks.Active || opts.Tasks.Inactive {
 			filter.Active = &opts.Tasks.Active
 		}
 
 		tasks, err := c.ListTasks(tenant, filter)
 		bail(err)
 
-		if opts.JSON {
-			fmt.Printf("%s\n", asJSON(tasks))
-			break
+		if opts.Tasks.Limit > 30 {
+			for i := 0; i < opts.Tasks.Limit/30; i++ {
+				filter := &shield.TaskFilter{
+					Status:            opts.Tasks.Status,
+					Limit:             &opts.Tasks.Limit,
+					Target:            opts.Tasks.Target,
+					BeforeCreatedDate: timeBefore,
+				}
+
+				if opts.Tasks.Active || opts.Tasks.Inactive {
+					filter.Active = &opts.Tasks.Active
+				}
+
+				sometasks, err := c.ListTasks(tenant, filter)
+				bail(err)
+				//current request smaller than API limit means we're EoDB
+				if len(sometasks) < 30 {
+					break
+				}
+
+				tasks = append(tasks, sometasks...)
+				timeBefore = tasks[len(tasks)-1].CreatedAt
+				time.Sleep(100 * time.Millisecond)
+			}
 		}
 
 		/* FIXME: support --long / -l and maybe --output / -o "fmt-str" */
-		tbl := tui.NewTable("UUID", "Type", "Status", "Owner", "Started at", "Completed at")
+		tbl := tui.NewTable("UUID", "Type", "Status", "Owner", "Created at", "Started at", "Completed at")
```

Comment:
I believe in `--json` format, this is `requested_at`, so we should make that match this column header, or make the column header match the JSON output.  Investigate and fix if necessary.